# **15** Let Definitions

_Definitions_ let us give a name to a value. This is similar to the concept of a variable in other programming languages, but it can't change.

In [1]:
let x = 42;;

val x : int = 42


In [2]:
x;;

- : int = 42


The `val x : int = 42` means we have a value, named `x`, set to the `int` with value `42`. The second expression simply evaluates to the `int` `42`.

We can also specify a type:

In [3]:
let y : int = 3110;;

val y : int = 3110


We can also use these values!

In [4]:
x + y;;

- : int = 3152


Hey! CS 3152 is "Introduction to Computer Game Design!"

---
Expressions and definitions are completely different.

`let` definitions have the syntax

```ocaml
let x = e
```
where `x` is an _identifier_, and `e` is an expression.

It is evaluated:
1. Evaluate `e` to value `v`.
2. Bind `v` to `x`: henceforth, `x` will evaluate to `v`.

`let` defintions **do not have a value themselves. They are not expressions.**

For example, this will fail:

In [5]:
(let x = 42) + 1;;

error: compile_error